doc/source/configuration/kayobe-automation.rst

```diff
@@ -0,0 +1,125 @@
+
+4. Provide all the relevant Kayobe :code:`group_vars` for :code:`github-runners` under :code:`inventory/hosts/github-runners`
+    * `infra-vms` ensuring all required `infra_vm_extra_network_interface` are defined
+    * `network-interfaces`
+    * `python-interpreter.yml` ensuring that `ansible_python_interpreter: /usr/bin/python3` has been set
+
+5. Create `runner.yml` file which will contain the variables required to deploy a series of runners
+
+.. code-block:: yaml
+
+    ---
+    runner_user: VM_USER_NAME_HERE
+    github_account: ORG_NAME_HERE
+    github_repo: KAYOBE_CONFIG_REPO_NAME_HERE
+    access_token: "{{ secrets_github_access_token }}"
+
+    base_runner_dir: /opt/actions-runner
+
+    default_runner_labels:
+      - kayobe
+      - openstack
+
+    github_runners:
+      runner_01: {}
+      runner_02: {}
+      runner_03: {}
+
+    docker_users:
+      - "{{ runner_user }}"
+
+    pip_install_packages:
+      - name: docker
+
+If using multiple environments add an extra label to :code:`default_runner_labels` to distinguish these runners from runners belonging to other environments.
+Also feel free to change the number of runners and their names.
+
+6. Obtain a personal access token that would enable the registration of GitHub runners against the `github_account` and `github_repo` defined above.
+    This token ideally should be `fine-grain personal access token <https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token>`__ which may require the organisation to enable such tokens beforehand steps can be found `here <https://docs.github.com/en/organizations/managing-programmatic-access-to-your-organization/setting-a-personal-access-token-policy-for-your-organization>`__
+    The repository permissions for a fine-grain personal access token should be; :code:`Actions: R/W, Administration: R/W, Metadata: R`
+    Once obtained the key add to :code:`secrets.yml` under :code:`secrets_github_access_token`
+
+7. Run :code:`kayobe playbook run ${KAYOBE_CONFIG_PATH}/ansible/deploy-github-runner.yml`
+
+8. Check runners have registered properly by visiting the repositories :code:`Action` tab -> :code:`Runners` -> :code:`Self-hosted runners`
+
+9. Repeat the above steps for each environment you intend to deploy runners within.
+    You can share the fine-grain access token between environments.
+
+Workflow Deployment
+-------------------
+
+1. Edit `inventory/group_vars/github-writer/writer.yml` in the base configuration making the appropriate changes to your deployments specific needs see documentation for `stackhpc.kayobe_workflows.github <https://github.com/stackhpc/ansible-collection-kayobe-workflows/tree/main/roles/github>`__
+
+2. Run :code:`kayobe playbook run ${KAYOBE_CONFIG_PATH}/ansible/write-workflows.yml`
+
+3. Add all required secrets to repository either via the GitHub UI or GitHub CLI (may require repository owner)
```

Comment:
A description of what these are would help here

Reply:
Description added.